libvast/vast/detail/logger_formatters.hpp

```diff
@@ -0,0 +1,486 @@
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const vast::record_type& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+template <>
+struct fmt::formatter<vast::data_view> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const vast::data_view& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+template <>
+struct fmt::formatter<vast::expression> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const vast::expression& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+// template <class... Sigs>
+// class typed_event_based_actor
+template <class... Sigs>
+struct fmt::formatter<caf::typed_event_based_actor<Sigs...>*> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const caf::typed_event_based_actor<Sigs...>* item,
+              FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", item->name());
+  }
+};
+
+// ---------std types
+
+template <typename T>
+struct fmt::formatter<std::vector<T>> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const std::vector<T>& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+template <>
+struct fmt::formatter<std::map<vast::offset, std::pair<size_t, vast::ids>>> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const std::map<vast::offset, std::pair<size_t, vast::ids>>& item,
+              FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+// ---------caf types
+
+template <class T>
+struct fmt::formatter<caf::optional<T>> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const caf::optional<T>& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
+  }
+};
+
+template <class T>
+struct fmt::formatter<caf::expected<T>> {
+  template <typename ParseContext>
+  constexpr auto parse(ParseContext& ctx) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  auto format(const caf::expected<T>& item, FormatContext& ctx) {
+    return format_to(ctx.out(), "{}", caf::deep_to_string(item));
```

Comment:
Shouldn't this go through the formatter for T instead of `caf::deep_to_string`? The same goes for `caf::optional<T>`.

Reply:
```suggestion
    if (item)
      return format_to(ctx.out(), "{}", *item);
    else
      return format_to(ctx.out(), "{}", item.error());
```

Reply:
I think we'd have to agree first how we want to format the nonopt/error case. Let's do this separately.

Reply:
Fine by me.